Sources/Build/PkgConfig.swift

```diff
@@ -0,0 +1,161 @@
+                              ]
+    
+    let name: String
+    let pcFile: String
+    var cFlags = [String]()
+    var libs = [String]()
+    private var parser: PkgConfigParser
+    
+    init(name: String) throws {
+        self.name = name
+        self.pcFile = try PkgConfig.locatePCFile(name: name)
+        parser = PkgConfigParser(pcFile: pcFile)
+    }
+    
+    static var envSearchPaths: [String] {
+        if let configPath = getenv("PKG_CONFIG_PATH") {
+            return configPath.characters.split(separator: ":").map(String.init)
+        }
+        return []
+    }
+    
+    static func locatePCFile(name: String) throws -> String {
+        for path in (searchPaths + envSearchPaths) {
+            let pcFile = Path.join(path, "\(name).pc")
+            if pcFile.isFile {
+                return pcFile
+            }
+        }
+        throw PkgConfigError.CouldNotFindConfigFile
+    }
+    
+    mutating func load() throws {
+        cFlags = [String]()
+        libs = [String]()
+        try parser.parse()
+        if let cFlags = parser.cFlags {
+            // FIXME: handle spaces in paths.
+            self.cFlags += cFlags.characters.split(separator: " ").map(String.init)
+        }
+        if let libs = parser.libs {
+            // FIXME: handle spaces in paths.
+            self.libs += libs.characters.split(separator: " ").map(String.init)
+        }
+        
+        if(parser.dependencies.isEmpty) {
+            return
+        }
+        
+        for dep in parser.dependencies {
+            var pkg = try PkgConfig(name: dep)
+            try pkg.load()
+            self.cFlags += pkg.cFlags
+            self.libs += pkg.libs
+        }
+    }
+}
+
+private struct PkgConfigParser {
+    let pcFile: String
+    var variables = [String: String]()
+    var dependencies = [String]()
+    var cFlags: String?
+    var libs: String?
+    
+    enum Token {
+        case CFlats(String)
+        case Libs(String)
+        case Dependencies(String)
+        case Variable(name: String, value: String)
+    }
+    
+    init(pcFile: String) {
+        self.pcFile = pcFile
+    }
+    
+    mutating func parse() throws {
+        let file = File(path: self.pcFile)
+        for line in try file.enumerate() {
+            if !line.characters.contains(":") && line.characters.contains("=")  {
+                let equalsIndex = line.characters.index(of: "=")!
+                let name = line[line.startIndex..<equalsIndex]
+                let value = line[equalsIndex.successor()..<line.endIndex]
+                variables[name] = resolveVariables(value)
+            } else if line.hasPrefix("Requires: ") {
+                dependencies = parseDependencies(value(line: line))
+            } else if line.hasPrefix("Libs: ") {
+                libs = resolveVariables(value(line: line)).chomp()
+            } else if line.hasPrefix("Cflags: ") {
+                cFlags = resolveVariables( value(line: line)).chomp()
+            }
```

Comment:
Is there a complete list available? If so, we could just check if it is in the known complete list and ignore if not there. 

We can at least partition into the following three cases, right?
1. Variable assignment
2. Key-value binding (Name: Value)
3. Comment